page.go

```diff
@@ -39,23 +42,23 @@ type page struct {
 	// case it is also the last page.
 	nextPage *page // Gets marshalled as nextPage.offset.
 
-	// payload contains the marshalled update, which may be spread over multiple
-	// pages if it is large. If spread over multiple pages, the full payload
-	// can be assembled by appending the separate payloads together. To keep the
-	// full size of the page at under pageSize bytes, the payload should not
-	// be greater than (pageSize - 64 bytes).
-	payload []byte // Gets marshalled to disk.
-
 	// transactionNumber is only saved for the last page, which can be
-	// determined by pageStatus. This number indicates the order in which this
-	// transaction was comitted.
+	// determined by nextPage being nil, or when marshalled to disk if
+	// it is marshalled to math.MaxUint64
 	transactionNumber uint64 // Gets marshalled to disk.
 
 	// transactionChecksum is the hash of all the pages and data in the
 	// committed transaction, including page status, nextPage, the payloads,
 	// and the transaction number. The checksum is only used internal to the
 	// WAL.
-	transactionChecksum [32]byte // Gets marshalled to disk for the last page only.
+	transactionChecksum [crypto.HashSize]byte // Gets marshalled to disk for the first page only.
```

Comment:
though it technically doesn't matter, I think logically you'd expect to see the checksum as either the first thing or the last thing. Since we are choosing to put the checksum on the first page (for performance), it makes sense that the checksum would be listed first.

Reply:
I guess technically you would want to put it above the first marshalled data, but below any fields that don't get marshalled (namely the offset)

Reply:
The little comment about being marshalled to disk for the first page only is incorrect, I think we marshal it to disk for all transactions, and then the payload just ends up being 4032 bytes per page.